packages/components/pie-link/src/defs.ts

```diff
@@ -1,8 +1,14 @@
 export const variants = ['default', 'high-visibility', 'inverse'] as const;
 export const sizes = ['small', 'medium'] as const;
 export const iconPlacements = ['leading', 'trailing'] as const;
+export const tags = ['a', 'button'] as const;
+export const buttonTypes = ['submit', 'button', 'reset', 'menu'] as const;
 
 export interface LinkProps {
+     /**
+     * What HTML element the link should be such as a or button.
+     */
+    tag: typeof tags[number];
```

Comment:
could we perhaps use an `isButton` prop instead?

Reply:
I don't mind changing to that but the team agreed to follow the generic approach of specifying tags during refinement and GitHub discussion.

Reply:
Fully appreciate that, it just feels more complicated and I'm not sure I understand what benefits it gives us over a boolean

Reply:
in case it clears a bit, the reasoning behind using `tag` instead of a boolean is to provide enough flexibility in cases we wanna render something else in the future other than a button or an anchor such as passing a framework-specific navigation element as a base for our component.

for example, Nextjs provides its own link component for in-app routing and you can't just use a native anchor tag to do that. So if our component is to be used there, we need to support something similar to what chakra-ui does [here](https://chakra-ui.com/docs/components/link/usage#usage-with-nextjs) 

I understand it might be less complex if we refactor to boolean, but passing `tag`, `as`, `renderAs` or `component` is quite common for components libraries.

Reply:
@raoufswe that makes sense, I kind of see it as a case of [YAGNI](https://martinfowler.com/bliki/Yagni.html) to make it this flexible prematurely.

I don't feel super strongly about this but I wanted to highlight I think it could be simpler :)

Reply:
`isButton` sounds for sure easier to implement and kind of easier for consumers to understand (in my opinion). But I also could se we needing to use different tags in other components, such as the `header` where per a11y there might be the need to render an `h1` style on an `h2` tag. We may want to keep the `tag` prop so that its use is consistent among components?

Reply:
Fair enough! PR looks good 👍 🎉